docs/rfcs/xxxx-integrations-in-operator.md

```diff
@@ -0,0 +1,265 @@
+The `type` field is an object with the following fields:
+
+* `allNodes`: True when the `name` integration should run on all Kubernetes
+  Nodes.
+* `unique`: True when the `name` integration must be unique across a
+  GrafanaAgent resource hierarchy.
+
+> Example of a valid integration:
+>
+> ```yaml
+> apiVersion: monitoring.grafana.com/v1alpha1
+> kind: Integration
+> metadata:
+>   name: mysql
+>   namespace: default
+> spec:
+>   name: mysqld_exporter
+>   type:
+>     allNodes: false # optional; false is default
+>     unique:   false # optional; false is default
+>   config:
+>     data_source_name: root@(mysql.default:3306)/
+>     disable_collectors: [slave_status]
+> ```
+
+GrafanaAgent will be updated to discover Integrations as part of its resource
+hierarchy. During reconcile, the following Kubernetes objects will be deployed:
+
+* One DaemonSet and Secret if there is at least one integration in the resource
+  hierarchy where `type.allNodes` is true.
+
+* One Deployment and Secret if there is at least one integration in the
+  resource hierarchy where `type.allNodes` is false.
+
+Secrets hold the generated Grafana Agent configuration; a Secret is used as
+integration configs may contain credentials.
+
+**NOTE**: As this functionality depends on [#1198][], integration pods will
+always be deployed with the experimental feature flag
+`-enable-feature=integrations-next` enabled. This also means that operator
+support for integrations requires a release of the agent where that
+experimental feature is available.
+
+### Integration validation
+
+The initial implementation of integrations support will have no knowledge of
+what integrations exist. As a result, the `spec.type` and `spec.config` fields
+for an Integration MUST be configured correctly for an integration to work.
+Users must refer to documentation to discover how `type` should be configured
+for their specific integration, and what settings are valid for the `config`
+block. Configuration errors will only surface as runtime errors from the
+deployed agent.
+
+Future versions of the Operator may:
+
+* Add knowledge for some integrations and validate `type` and `config`
+  accordingly (though breaking changes to the config at the Agent level may
+  introduce extra complexity to this).
+
+* Update the `status` field of the root GrafanaAgent resource during reconcile
+  to expose any reconcile or runtime errors.
+
+### Additional settings for the Integration CRD
+
+Some integrations may require changes to the deployed Pods to function
+properly. Integrations will additionally support declaring `volumes`,
+`volumeMounts`, `secrets` and `configMaps`. These fields will be merged with
+the fields of the same name from the root GrafanaAgent resource when creating
+integration pods:
+
+> ```yaml
+> apiVersion: monitoring.grafana.com/v1alpha1
+> kind: Integration
+> metadata:
+>   name: kafka
+>   namespace: default
+> spec:
+>   name: kafka_exporter
+>   type:
+>     hasMetrics: true
```

Comment:
Is this another `type` field?

Reply:
Relic of an older revision, this should be removed, sorry :)